hnn_core/viz.py

```diff
@@ -252,3 +292,141 @@ def plot_cells(net, ax=None, show=True):
         plt.show()
 
     return ax.get_figure()
+
+
+def plot_tfr_morlet(dpl, *, freqs, n_cycles=7., tmin=None, tmax=None,
+                    layer='agg', decim=False, ax=None,
+                    colorbar=True, show=True):
+    """Plot Morlet time-frequency representation of dipole time course
+
+    Parameters
+    ----------
+    dpl : instance of Dipole | list of Dipole instances
+        The Dipole object.
+    fregs : array
+        Frequency range of interest.
+    n_cycles : float | array of float, default 7.0
+        Number of cycles. Fixed number or one per frequency.
+    tmin : float or None
+        Start time of plot in milliseconds. If None, plot entire simulation.
+    tmax : float or None
+        End time of plot in milliseconds. If None, plot entire simulation.
+    layer : str, default 'agg'
+        The layer to plot. Can be one of 'agg', 'L2', and 'L5'
+    decim : int or None
+        Factor by which to decimate the raw dipole traces (optional)
+    ax : instance of matplotlib figure | None
+        The matplotlib axis
+    colorbar : bool
+        If True (default), adjust figure to include colorbar.
+    show : bool
+        If True, show the figure
+
+    Returns
+    -------
+    fig : instance of matplotlib Figure
+        The matplotlib figure handle.
+    """
+
+    import matplotlib.pyplot as plt
+    from matplotlib.ticker import ScalarFormatter
+    from mne.time_frequency import tfr_array_morlet
+
+    data, times = _get_plot_data(dpl, layer, tmin, tmax)
+
+    sfreq = dpl.sfreq
+    if decim is not None:
+        data, times = _decimate_plot_data(decim, data, times)
+        sfreq = sfreq / decim
+
+    # mirror padding!
+    data = np.r_[data[-1:0:-1], data, data[-2::-1]]
```

Comment:
there seems to be a strong edge effect now!

Reply:
I noticed that, should I zero-pad instead?

Reply:
maybe give it a quick try? Let's not be too distracted by this

Reply:
I've added zero-padding as the default, arriving in the next commit. Looks better than with mirroring for this dataset.